api-lib/sendSocialMessage.ts

```diff
@@ -0,0 +1,88 @@
+import fetch from 'node-fetch';
+
+import { gql } from './Gql';
+
+const DISCORD_BOT_NAME = 'Caesar The Coordinape Bot';
+const DISCORD_BOT_AVATAR_URL =
+  'https://coordinape-prod.s3.amazonaws.com/assets/static/20220131-coordinape-pfp-ape.jpg';
+
+type SocialMessage = {
+  message: string;
+  circleId: number;
+  sanitize: boolean;
+  channels: {
+    discord?: boolean;
+    telegram?: boolean;
+  };
+};
+
+function cleanStr(str: string) {
+  return str.replace(/:|-|\/|\*|_|`/g, '');
+}
+
+export async function sendSocialMessage({
+  message,
+  circleId,
+  sanitize = true,
+  channels,
+}: SocialMessage) {
+  const msg = sanitize ? cleanStr(message) : message;
+
+  const { circles } = await gql.q('query')({
```

Comment:
see how Zashton moved these to `import { Gql } from '../src/lib/gql/GqlHasuraAdmin';`
and actually it doesn't need to be admin, so just regular '../src/lib/gql';`

Reply:
I see, do you want these exact queries into `../src/lib/gql` or more general ones? In this case the more general one would be e.g. `getCircleById`?

Reply:
Btw, `../src/lib/gql` doesn't export an api object, but rather the comment in it tells to do this:

```
import { REACT_APP_HASURA_URL } from 'config/env';
import { getAuthToken } from 'services/api';
import { getGql } from 'lib/gql';

const api = getApi(REACT_APP_HASURA_URL, getAuthToken);
await api.updateProfile();
```

It seems like it'd be easier to export the `api` to avoid having to refer to `REACT_APP_HASURA_URL` etc. all over app code.

Reply:
Auth token isn't available until after the user logs in

Reply:
I think that comment is confusing, rather, I think we ought to link to the usages..

But like lawrence says this instruction is for the react app.

For the vercel functions, see`api-lib/Gql.ts`
which has the admin client.